api/src/main/java/io/serverlessworkflow/api/deserializers/DataInputSchemaDeserializer.java

```diff
@@ -46,17 +53,44 @@ public DataInputSchema deserialize(JsonParser jp, DeserializationContext ctxt)
     JsonNode node = jp.getCodec().readTree(jp);
 
     DataInputSchema dataInputSchema = new DataInputSchema();
+    JsonNode schemaDefinition = null;
 
-    if (!node.isObject()) {
-      dataInputSchema.setSchema(node.asText());
-      dataInputSchema.setFailOnValidationErrors(true); // default
-
-      return dataInputSchema;
-    } else {
-      dataInputSchema.setSchema(node.get("schema").asText());
+    if (node.isObject() && node.get("schema").isObject() && !node.get("schema").isEmpty()) {
+      schemaDefinition = node.get("schema");
       dataInputSchema.setFailOnValidationErrors(node.get("failOnValidationErrors").asBoolean());
+    } else {
+      String schemaFileDef = node.isObject() ? node.get("schema").asText() : node.asText();
+      dataInputSchema.setFailOnValidationErrors(true);
+      dataInputSchema.setRefValue(schemaFileDef);
+      String schemaFileContent = Utils.getResourceFileAsString(schemaFileDef);
+      JsonNode schemaRefNode;
+      ObjectMapper jsonWriter = new ObjectMapper();
+      if (schemaFileContent != null && schemaFileContent.trim().length() > 0) {
+        // if its a yaml def convert to json first
+        if (!schemaFileContent.trim().startsWith("{")) {
+          // convert yaml to json to validate
+          ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+          Object schemaObj = yamlReader.readValue(schemaFileContent, Object.class);
+          schemaRefNode =
+              jsonWriter.readTree(
+                  new JSONObject(jsonWriter.writeValueAsString(schemaObj)).toString());
+        } else {
+          schemaRefNode = jsonWriter.readTree(new JSONObject(schemaFileContent).toString());
+        }
+
+        JsonNode schemaRef = schemaRefNode.get("schema");
```

Comment:
hmmmm, Im not sure this is right 
schemaRefNode is suppose to have the content of the file loaded in memory, which is something like this
```
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "Expression",
    "description": "Schema for expression test",
    "type": "object",
    "properties": {
        "numbers": {
            "description": "The array of numbers to be operated with",
            "type": "array",
            "items" : {
                "type" : "object",
                "properties" : {
                   "x" : {"type":"number"},
                   "y" : {"type":"number"}
                }
             }
        }
    },
    "required": ["numbers"]
}
```
so there is not "schema" field there, and I suspect the schemadefinition is always set to null when schemaFileDef is not null, making this whole loading code essentially dead code. 
You should be able to verify that through testing